ansible/roles/fanout/templates/arista_7060_deploy.j2

```diff
@@ -36,14 +36,18 @@ interface defaults
 {% for intf in device_port_vlans[inventory_hostname] %}
 interface {{ intf }}
    description {{ device_conn[inventory_hostname][intf]['peerdevice'] }}-{{ device_conn[inventory_hostname][intf]['peerport'] }}
+{%     if device_conn[inventory_hostname][intf]['autoneg']|lower == "on" %}
+   speed auto {{ device_conn[inventory_hostname][intf]['speed'] }}full
+{%   else %}
    speed force {{ device_conn[inventory_hostname][intf]['speed'] }}full
+{%     endif %}
 {%   if device_port_vlans[inventory_hostname][intf]['mode'] == 'Trunk' %}
    switchport mode trunk
    switchport trunk allowed vlan {{ device_port_vlans[inventory_hostname][intf]['vlanids'] }}
 {%   else %}
    switchport mode dot1q-tunnel
    switchport access vlan {{ device_port_vlans[inventory_hostname][intf]['vlanids'] }}
-{%     if device_conn[inventory_hostname][intf]['speed'] == "100000" %}
+{%     if device_conn[inventory_hostname][intf]['speed'] == "100000" and device_conn[inventory_hostname][intf]['autoneg']|lower == "off" %}
```

Comment:
@vdahiya12 Why are we restricting to 100G only?

Reply:
there is no restriction here, only if speed is 100000 and autoneg is off, then we keep fec as reed-solomon

Reply:
add the peer device check

Reply:
@vdahiya12 prevent configuration of RS FEC if msft_an_lt is enabled?